packages/constellation/src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable import/no-unresolved */
+import '../styles.css'
+
+import React, { useEffect, useState } from 'react'
+
+const ThemeProvider = ({
+  children,
+  theme,
+  colorOverride = {},
+}: {
+  children: React.ReactNode
+  colorOverride: {
+    primary?: string
+  }
+  theme: string
+}) => {
+  const [colorOverrideStyles, setColorOverrideStyles] = useState({})
```

Comment:
you can use the useMemo hook to recreate the colorOverideStyles object every time the colorOveride changes.
and use reduce to create the object you want
like this

```
const colorOverrideStyles = useMemo((): CSSProperties => {
  const colors: keyof typeof colorOverride = Object.keys(colorOverride)

  return colors.reduce((newColorOverride, color) => ({
    ...newColorOverride,
    [`--color-${color}`]: colorOverride[color],
  }), {})
}, [colorOverride])
```

and maybe you could move the body of the memo to a utility file and then just call
```
const colorOverrideStyles = useMemo(() => transformColorOveride(colorOverride), [colorOverride])
```

What you think?

Reply:
And do you think this transformation from `colorOverride` to CSS properties could have a unit test